omniledger/service/service.go

```diff
@@ -54,6 +54,10 @@ type Service struct {
 	// collections cannot be stored, so they will be re-created whenever the
 	// service reloads.
 	collectionDB map[string]*collectionDB
+	// holds a link for every omniledger to the latest block that is included in
+	// the collection.
+	// TODO: merge collectionDB and pollChan into olState structure.
+	state olState
```

Comment:
I'm not sure whether we need a separate field. Most of the fields in this struct are states anyway. Further, `olState` has a mutex on it. If everything is sharing the same mutex, that'd hinder performance.

Reply:
it's mostly read/write mutex for a shared resource - I don't think it'll be a big performance issue.